### PR TITLE
the scanner is now less eager about deprecations

### PR DESCRIPTION
**Pull request by xeno-by**
When healing braces it isn't very useful to report deprecation warnings,
especially since this process is just simple context-free skimming, which
can't know about what positions can accept what identifiers.

### CODE REVIEW

**Comment by xeno-by** [comment]
Supersedes https://github.com/scala/scala/pull/1763

**Comment by xeno-by** [comment]
review @paulp 

btw, in a totally random fashion I stumbled upon a repro for the issue fixed in this patch :)

File: test/files/neg/macro-false-deprecation-warning.flags [hunk 1]
```diff
@@ -0,0 +1 @@
+-language:experimental.macros
```
Comment (paulp) [inline test/files/neg/macro-false-deprecation-warning.flags]: The test doesn't test anything unless "-deprecation" is given. (I tried compiling the test case without it and it produces the checkfile, and doesn't even say "there were deprecation warnings.")

Comment (xeno-by) [inline test/files/neg/macro-false-deprecation-warning.flags]: Here's what I get when commenting out the `override def emitIdentifierDeprecationWarnings = false`:

```
22:19 ~/Projects/Kepler_ident-deprecation-warnings/test (topic/ident-deprecation-warnings)$ ./partest files/neg/macro-false-deprecation-warning

Testing individual files
testing: [...]/files/neg/macro-false-deprecation-warning              [FAILED]
1,3d0
< Impls_Macros_1.scala:13: warning: macro is now a reserved word; usage as an identifier is deprecated
<     def unapplySeq[T](x: List[T]): Option[Seq[T]] = macro impl[T]
<                                                     ^
7d3
< one warning found
1 of 1 tests failed (elapsed time: 00:00:01)
```

Comment (xeno-by) [inline test/files/neg/macro-false-deprecation-warning.flags]: Omg Paul I thought I was going mad, but apparently it's partest who silently enables `-deprecation` for all tests:

https://github.com/scalamacros/kepler/blob/e5d34d70499504e085ddf957c1c818ffb63f4e8d/src/partest/scala/tools/partest/nest/CompileManager.scala#L44

```
class TestSettings(cp: String, error: String => Unit) extends Settings(error) {
  def this(cp: String) = this(cp, _ => ())

  deprecation.value = true
  nowarnings.value  = false
  encoding.value    = "UTF-8"
  classpath.value   = cp
}
```

Comment (paulp) [inline test/files/neg/macro-false-deprecation-warning.flags]: Nice. I knew about that (not why, only that it exists) but forgot. #1824.

**Comment by xeno-by** [comment]
PLS REBUILD ALL

**Comment by paulp** [comment]
Still a little lukewarm, but lgtm.

**Pull request merged**